microbench/src/main/java/io/netty/microbench/handler/ssl/AbstractSslEngineBenchmark.java

```diff
@@ -0,0 +1,281 @@
+
+        serverAppReadBuffer = allocateBuffer(
+                serverEngine.getSession().getApplicationBufferSize());
+        clientAppReadBuffer = allocateBuffer(
+                clientEngine.getSession().getApplicationBufferSize());
+        empty = allocateBuffer(0);
+    }
+
+    protected final void destroyHandshakeBuffers() {
+        freeBuffer(cTOs);
+        freeBuffer(sTOc);
+        freeBuffer(serverAppReadBuffer);
+        freeBuffer(clientAppReadBuffer);
+        freeBuffer(empty);
+    }
+
+    protected final boolean doHandshake() throws SSLException {
+        clientEngine.beginHandshake();
+        serverEngine.beginHandshake();
+
+        SSLEngineResult clientResult = null;
+        SSLEngineResult serverResult = null;
+
+        boolean clientHandshakeFinished = false;
+        boolean serverHandshakeFinished = false;
+
+        do {
+            int cTOsPos = cTOs.position();
+            int sTOcPos = sTOc.position();
+
+            if (!clientHandshakeFinished) {
+                clientResult = clientEngine.wrap(empty, cTOs);
+                runDelegatedTasks(clientResult, clientEngine);
+                assert empty.remaining() == clientResult.bytesConsumed();
+                assert cTOs.position() - cTOsPos == clientResult.bytesProduced();
+
+                if (isHandshakeFinished(clientResult)) {
+                    clientHandshakeFinished = true;
+                }
+            }
+
+            if (!serverHandshakeFinished) {
+                serverResult = serverEngine.wrap(empty, sTOc);
+                runDelegatedTasks(serverResult, serverEngine);
+                assert empty.remaining() == serverResult.bytesConsumed();
+                assert sTOc.position() - sTOcPos == serverResult.bytesProduced();
+
+                if (isHandshakeFinished(serverResult)) {
+                    serverHandshakeFinished = true;
+                }
+            }
+
+            cTOs.flip();
+            sTOc.flip();
+
+            cTOsPos = cTOs.position();
+            sTOcPos = sTOc.position();
+
+            if (!clientHandshakeFinished) {
+                int clientAppReadBufferPos = clientAppReadBuffer.position();
+                clientResult = clientEngine.unwrap(sTOc, clientAppReadBuffer);
+
+                runDelegatedTasks(clientResult, clientEngine);
+                assert sTOc.position() - sTOcPos == clientResult.bytesConsumed();
+                assert clientAppReadBuffer.position() - clientAppReadBufferPos == clientResult.bytesProduced();
+
+                if (isHandshakeFinished(clientResult)) {
+                    clientHandshakeFinished = true;
+                }
+            } else {
+                assert !sTOc.hasRemaining();
+            }
+
+            if (!serverHandshakeFinished) {
+                int serverAppReadBufferPos = serverAppReadBuffer.position();
+                serverResult = serverEngine.unwrap(cTOs, serverAppReadBuffer);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "serverAppReadBufferPos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)